cmake/MakeLibrary.cmake

```diff
@@ -33,98 +35,93 @@ macro(make_target_impl NAME FLAGS REQUIREMENTS IS_LIBRARY SHARED SHARED_ALL_DEFI
 
   # add a custom property to the target listing its dependencies
   if(NOT ${FLAGS} STREQUAL "")
-    set_property(TARGET ${NAME} APPEND_STRING PROPERTY COMPILE_FLAGS " ${FLAGS}")
+    set_property(TARGET ${NAME} APPEND_STRING PROPERTY COMPILE_FLAGS
+                                                       " ${FLAGS}")
   endif()
-  if (${IS_LIBRARY})
-    if (NOT WIN32)
-      #windows is always fPIC
+  if(${IS_LIBRARY})
+    if(NOT WIN32)
+      # windows is always fPIC
       set_property(TARGET ${NAME} APPEND_STRING PROPERTY COMPILE_FLAGS " -fPIC")
     endif()
-    if (APPLE)
-      if (${SHARED})
-        if (NOT ${SHARED_ALL_DEFINED})
-          set_property(TARGET ${NAME} APPEND_STRING PROPERTY LINK_FLAGS " -undefined dynamic_lookup")
+    if(APPLE)
+      if(${SHARED})
+        if(NOT ${SHARED_ALL_DEFINED})
+          set_property(TARGET ${NAME} APPEND_STRING
+                       PROPERTY LINK_FLAGS " -undefined dynamic_lookup")
         endif()
       endif()
     endif()
   endif()
 
-  if (${IS_LIBRARY})
+  if(${IS_LIBRARY})
     if(${SHARED})
       target_link_libraries(${NAME} PRIVATE ${REQUIREMENTS})
     elseif(${OBJECT})
       # TODO we can link the requirements from here when target_link_libraries
-      # works with OBJECT library targets (requires CMake 3.12)
-      # See https://gitlab.kitware.com/cmake/cmake/issues/14778
-      # For now, do nothing.
+      # works with OBJECT library targets (requires CMake 3.12) See
+      # https://gitlab.kitware.com/cmake/cmake/issues/14778 For now, do nothing.
     else()
       target_link_libraries(${NAME} PUBLIC ${REQUIREMENTS})
     endif()
   else()
     target_link_libraries(${NAME} PUBLIC ${REQUIREMENTS})
   endif()
-    
-  # Ensure dependencies are tracked in order to make sure compilation order matters.
-  add_dependencies(${NAME} "${REQUIREMENTS}")
-    
+
+  # Ensure dependencies are tracked in order to make sure compilation order
+  # matters.
+  if (REQUIREMENTS)
+      add_dependencies(${NAME} ${REQUIREMENTS})
+  endif()
+
   # make sure dependencies are always built first
-  add_dependencies(${NAME} "${_TC_EXTERNAL_DEPENDENCIES}")
-  add_dependencies(${NAME} external_dependencies)
-endmacro()
+  if (TC_EXTERNAL_DEPENDENCIES)
+    add_dependencies(${NAME} ${TC_EXTERNAL_DEPENDENCIES})
+  endif()
 
+  if (external_dependencies)
+    add_dependencies(${NAME} external_dependencies)
+  endif()
 
-# This is an external function
-# Usage:
-#    make_library(NAME target
-#                 SOURCES a.cpp b.cpp
-#                 REQUIRES libx liby
-#                 MAC_REQUIRES libz libzz
-#                 LINUX_REQUIRES libk libj
-#                 [SHARED] [OUTPUT_NAME xxxx] [SHARED_ALL_DEFINED]
-#                 [OBJECT]
-#                 )
-# Example:
+endmacro()
+
+# This is an external function Usage: make_library(NAME target SOURCES a.cpp
```

Comment:
The auto formatting messed up all the tables in the help -- maybe make the change without reformatting the file?  Or figure out how to get it to not mess these up?

Reply:
done.